routes/verify.go

```diff
@@ -251,6 +252,208 @@ func (fes *APIServer) validatePhoneNumberNotAlreadyInUse(phoneNumber string, use
+	if requestData.RewardNanos > fes.Config.StarterDESONanos {
+		_AddBadRequestError(ww, fmt.Sprintf("HandleAdminUpdateCaptchaRewardRequest: Reward amount %v exceeds starter deso amount %v", requestData.RewardNanos, fes.Config.StarterDESONanos))
+		return
+	}
+
+	if err := fes.putCaptchaRewardNanosInGlobalState(requestData.RewardNanos); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("HandleAdminUpdateCaptchaRewardRequest: Error putting captcha reward in global state: %v", err))
+		return
+	}
+
+	res := AdminUpdateCaptchaRewardResponse{
+		RewardNanos: requestData.RewardNanos,
+	}
+	if err := json.NewEncoder(ww).Encode(res); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("HandleAdminUpdateCaptchaRewardRequest: Problem encoding response: %v", err))
+		return
+	}
+}
+
+// getCaptchaRewardNanosFromGlobalState returns the amount of nanos to reward for a successful captcha from global state.
+func (fes *APIServer) getCaptchaRewardNanosFromGlobalState() (uint64, error) {
+	dbKey := GlobalStateKeyForCaptchaRewardAmountNanos()
+
+	rewardNanosBytes, err := fes.GlobalState.Get(dbKey)
+	if err != nil {
+		return 0, fmt.Errorf(
+			"getCaptchaRewardNanosFromGlobalState: Problem with Get: %v", err)
+	}
+
+	rewardNanos, err := lib.ReadUvarint(bytes.NewReader(rewardNanosBytes))
+
+	return rewardNanos, nil
+}
+
+// putCaptchaRewardNanosInGlobalState puts the amount of nanos to reward for a successful captcha in global state.
+func (fes *APIServer) putCaptchaRewardNanosInGlobalState(rewardNanos uint64) error {
+	dbKey := GlobalStateKeyForCaptchaRewardAmountNanos()
+
+	rewardNanosBytes := lib.UintToBuf(rewardNanos)
+
+	if err := fes.GlobalState.Put(dbKey, rewardNanosBytes); err != nil {
+		return fmt.Errorf(
+			"putCaptchaRewardNanosInGlobalState: Problem with Put: %v", err)
+	}
+
+	return nil
+}
+
+// verifyHCaptchaTokenAndSendStarterDESO verifies the captcha token and sends the starter DESO to the user.
+func (fes *APIServer) verifyHCaptchaTokenAndSendStarterDESO(token string, publicKeyBase58Check string) (txnHashHex string, err error) {
+	// Verify the token with hCaptcha.
+	verificationSuccess, err := fes.verifyHCaptchaToken(token)
+
+	if err != nil {
+		return "", fmt.Errorf("HandleCaptchaVerificationRequest: Error verifying captcha: %v", err)
+	}
+
+	if !verificationSuccess {
+		return "", fmt.Errorf("HandleCaptchaVerificationRequest: Captcha verification failed")
+	}
+
+	// Ensure the user has not already received the starter DESO for submitting a successful captcha.
+	userMetadata, err := fes.getUserMetadataFromGlobalState(publicKeyBase58Check)
+	if err != nil {
+		return "", fmt.Errorf("HandleCaptchaVerificationRequest: Problem with getUserMetadataFromGlobalState: %v", err)
+	}
+
+	if userMetadata.LastHcaptchaBlockHeight != 0 {
+		return "", fmt.Errorf("HandleCaptchaVerificationRequest: LastHcaptchaBlockHeight is already set")
+	}
+
+	if fes.Config.StarterDESOSeed == "" {
```

Comment:
we should move this check higher up since we don't need to verify the captcha or hit global state if we're in this state.